src/app/models/submission.server.model.ts

```diff
@@ -115,6 +118,51 @@ SubmissionSchema.statics.findFormsWithSubsAbove = function (
   ]).exec()
 }
 
+/**
+ * Creates a new email submission only if provided submitterId is unique.
+ * This method ensures that isSingleSubmission is enforced.
+ * @param submitterId uniquely identifies the submitter
+ * @returns created submission if successful, null otherwise
+ */
+SubmissionSchema.statics.saveIfSubmitterIdIsUnique = async function (
+  formId,
+  submitterId,
+  submissionContent,
+) {
+  const session = await this.startSession()
```

Comment:
@kevin9foong after a little more consideration I think we don't have to further optimise this to avoid the read, since it's more costly to `create` than to `read`. Thus there's some inherent benefits of keeping the `read`.

Reply:
yeah, this makes sense. the cost to abort a create may be higher than the cost to do a exist read check (aborting after a read should be minimal cost apart from txn overhead)